### ISSUE TITLE
Curate decompiler sidecar facts after recent raise wave

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1114]
## Goal

Run a small curator pass over decompiler sidecar facts after the recent raise wave. The central ledger and scorecard self-checks pass, but several sidecar `PositiveCoverage` / `AdversarialCoverage` / `MissingDiscriminator` strings appear stale relative to current `LoweringCoverage` and scorecard rows.

## Evidence

Checked on current `origin/main` (`b838dc07` during review):

```bash
dotnet run --project src/ILInspector.Decompiler.Tests -c Release -- \
  -class ILInspector.Decompiler.Tests.LoweringCoverageTests \
  -class ILInspector.Decompiler.Tests.LoweringFactCatalogTests \
  -class ILInspector.Decompiler.Tests.IdiomShapeScorecardTests
```

Result: 10 tests, 0 failed. Structural invariants hold.

But sidecar notes look stale in at least these areas:

- `TupleRecoveryFacts` / `TupleCreationExpression`: still says nested/rest tuples not recovered, while `LoweringCoverage` now says eight-or-more nested-TRest form is flattened to one literal and the scorecard includes `TupleRest`.
- `ForeachRecoveryFacts`: still says higher-rank arrays are not raised, while `LoweringCoverage` now says rank-N rectangular array lowering is covered.
- `IsPatternRecoveryFacts`: still broadly says list sub-patterns remain unraised, while `LoweringCoverage` now records the single-element string-array list pattern with constant OR alternatives.

## Scope

Metadata/text only:

- update sidecar `PositiveCoverage`, `AdversarialCoverage`, and `MissingDiscriminator` strings to match current implemented coverage;
- do not broaden or change any matcher;
- do not add scorecard rows except if a recovered positive already exists and the scorecard somehow missed it;
- keep negatives in pass-level tests, not the scorecard.

## Validation

Run:

```bash
dotnet run --project src/ILInspector.Decompiler.Tests -c Release -- \
  -class ILInspector.Decompiler.Tests.LoweringCoverageTests \
  -class ILInspector.Decompiler.Tests.LoweringFactCatalogTests \
  -class ILInspector.Decompiler.Tests.IdiomShapeScorecardTests
```

If markdown docs are touched, run markdownlint on changed docs.

**Comment by richlander** [comment]
I’ll take this curator pass.

**Comment by richlander** [comment]
Opened PR #1120 for this curator pass.

Updated sidecar fact text for:
- TupleCreationExpression: rest tuple positive/negative coverage and element-name-only remaining gap
- ForEachStatement: rank-N rectangular array coverage and narrowed missing discriminator
- IsPatternOperator: single-element string-array list-pattern positive/negative coverage and broader list-pattern frontier wording

Validation passed with the requested LoweringCoverage, LoweringFactCatalog, and IdiomShapeScorecard test classes.
